src/nsrt_learning.py

```diff
@@ -248,8 +244,102 @@ def _learn_pnad_side_predicates(
         ground_atom_dataset: List[GroundAtomTrajectory],
         train_tasks: List[Task], predicates: Set[Predicate],
         segments: List[Segment],
-        verbose: bool) -> List[PartialNSRTAndDatastore]:
-    raise NotImplementedError
+        segmented_trajs: List[List[Segment]]) -> List[PartialNSRTAndDatastore]:
+    # There are no goal states for this search; run until exhausted.
+    def _check_goal(s: Tuple[PartialNSRTAndDatastore, ...]) -> bool:
+        del s  # unused
+        return False
+
+    # Consider changing each effect to a side predicate.
+    def _get_successors(
```

Comment:
so this will be a search over PNADs and data stores? why are we doing it that way? i would've thought the search would just be over the operators themselves. is this an efficiency thing?

Reply:
it is over just the operators, i just keep them in the PNAD data structure because it's easier to write the code. it shouldn't matter from an efficiency perspective, it's not like the datastores themselves are getting copied or anything (there's no deep copying)